src/libraries/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -51,6 +52,37 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, int i
             return default;
         }
 
+        /// <summary>Returns the element at a specified index in a sequence.</summary>
+        /// <param name="source">An <see cref="IEnumerable{T}" /> to return an element from.</param>
+        /// <param name="index">The index of the element to retrieve, which is either from the start or the end.</param>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="index" /> is outside the bounds of the <paramref name="source" /> sequence.
+        /// </exception>
+        /// <returns>The element at the specified position in the <paramref name="source" /> sequence.</returns>
+        public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, Index index)
+        {
+            if (source == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
+            }
+
+            if (!index.IsFromEnd)
+            {
+                return source.ElementAt(index.Value);
+            }
+
+            if (TryGetElementAt(source, index.Value, out TSource? result))
+            {
+                return result;
+            }
+
+            ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
+            return default;
```

Comment:
```suggestion
            if (!TryGetElementAt(source, index.Value, out TSource? result))
            {
                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
            }

            return result;
```

Reply:
Updated. The 2 versions are the same. 

When the ThrowHelper call is at the bottom, it must have a return statement, because the ThrowHelper call wraps the `throw new Exception(...)` statement inside it. This problem shouldn't exist, if the helper method only wraps `new Exception(...)`:
```cs
throw ExceptionHelper.ArgumentOutOfRange(ExceptionArgument.index);
```